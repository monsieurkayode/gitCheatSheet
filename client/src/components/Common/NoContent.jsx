import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  text-align: center;
  max-width: 1232px;
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  flex: 1;
  width: 100%;
  transform: translateY(-64px);

  i {
    color: #4d75a1;
    font-size: 50px;
  }
  
  .no-content-info {
    color: #668090;
    font-family: Roboto, sans-serif;
    font-weight: 400;
  }
`;

const NoContent = ({ content }) => (
  <Container className="no-content">
    <i className="material-icons">info</i>
    <h3 className="no-content-info">
      {content}
    </h3>
  </Container>
);

NoContent.propTypes = {
  content: PropTypes.string.isRequired
};

export default NoContent;
