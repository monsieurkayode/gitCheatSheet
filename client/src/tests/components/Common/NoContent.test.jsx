import React from 'react';
import { mount } from 'enzyme';
import NoContent from '../../../components/Common/NoContent';

describe('NoContent Component', () => {
  it('should render correctly and does not crash', () => {
    const wrapper = mount(<NoContent content="No content found" />);
    expect(wrapper).toMatchSnapshot();
  });
});
