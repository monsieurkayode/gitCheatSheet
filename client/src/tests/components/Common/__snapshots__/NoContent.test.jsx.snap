// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoContent Component should render correctly and does not crash 1`] = `
<NoContent
  content="No content found"
>
  <styled.div
    className="no-content"
  >
    <StyledComponent
      className="no-content"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "hazPbA",
            "rules": Array [
              "
  text-align: center;
  max-width: 1232px;
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  flex: 1;
  width: 100%;
  transform: translateY(-64px);

  i {
    color: #4d75a1;
    font-size: 50px;
  }
  
  .no-content-info {
    color: #668090;
    font-family: Roboto, sans-serif;
    font-weight: 400;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="no-content sc-bdVaJa hazPbA"
      >
        <i
          className="material-icons"
        >
          info
        </i>
        <h3
          className="no-content-info"
        >
          No content found
        </h3>
      </div>
    </StyledComponent>
  </styled.div>
</NoContent>
`;
